### PR TITLE
ci: create update-data scheduled workflow

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Adds a scheduled, and on demand, GitHub Actions workflow to update static data in the Renovate repository.

TODO:
- [x] Document this workflow
- [x] Test this workflow
	- https://github.com/JamieMagee/renovate/pull/2

## Context

#17050

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
we should customize commit and PR message to follow semver fix release

File: .github/workflows/update-data.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Update data
+        run: yarn run update-static-data
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@923ad837f191474af6b1721408744feb989a4c27 # tag=v4.0.4
+        with:
+          author: 'Renovate Bot <renovate@whitesourcesoftware.com>'
+          branch: 'chore/update-static-data'
+          commit-message: 'ci(data): automatic update of static data'
```
Comment (viceice) [inline .github/workflows/update-data.yml:37]:
```suggestion
          commit-message: 'fix(data): automatic update of static data'
```
so we get a new release

File: .github/workflows/update-data.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          committer: 'Renovate Bot <renovate@whitesourcesoftware.com>'
+          title: 'ci(data): automatic update of static data'
```
Comment (viceice) [inline .github/workflows/update-data.yml:39]:
```suggestion
          title: 'fix(data): automatic update of static data'
```

File: .github/workflows/update-data.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: 'Update static data'
+on:
+  schedule:
+    - cron: '0 0 * * 0' # Every Sunday at midnight
+  workflow_dispatch:
+
+env:
+  NODE_VERSION: 14
```
Comment (HonkingGoose) [inline .github/workflows/update-data.yml:8]:
```suggestion
  NODE_VERSION: 16
```

Maybe update to node 16?

Comment (JamieMagee) [reply .github/workflows/update-data.yml:8]: Sure! I only chose 14 as it's what all other workflows use.

**Review by viceice — APPROVED** [review]

**Pull request merged**